src/AzureAIProxy/Routes/AzureOpenAIAssistants.cs

```diff
@@ -0,0 +1,167 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using AzureAIProxy.Shared.Database;
+using AzureAIProxy.Routes.CustomResults;
+using AzureAIProxy.Services;
+
+namespace AzureAIProxy.Routes;
+
+/// <summary>
+/// Defines routes and handling logic for interacting with Azure OpenAI assistants and threads.
+/// </summary>
+public static class AzureAIOpenAIAssistants
+{
+    /// <summary>
+    /// Maps routes for assistant and thread operations under the "/openai" path.
+    /// </summary>
+    /// <param name="builder">The route group builder to configure the routes.</param>
+    /// <returns>The updated route group builder.</returns>
+    public static RouteGroupBuilder MapAzureOpenAIAssistantsRoutes(this RouteGroupBuilder builder)
+    {
+        var openAiPaths = new[] { "/assistants/{*assistant_id}", "/threads/{*thread_id}" };
+        var openAIGroup = builder.MapGroup("/openai");
+
+        foreach (var path in openAiPaths)
+        {
+            openAIGroup.MapPost(path, CreateThreadAsync);
+            openAIGroup.MapGet(path, CreateThreadAsync);
+            openAIGroup.MapDelete(path, CreateThreadAsync);
+        }
+        return builder;
+    }
+
+    /// <summary>
+    /// Handles HTTP requests for assistant and thread operations by routing them to the appropriate method based on the request type.
+    /// </summary>
+    /// <param name="catalogService">The catalog service for retrieving deployment information.</param>
+    /// <param name="proxyService">The proxy service for forwarding requests.</param>
+    /// <param name="assistantService">The assistant service for managing assistant and thread IDs.</param>
+    /// <param name="context">The HTTP context of the request.</param>
+    /// <param name="requestJsonDoc">The optional JSON document in the request body.</param>
+    /// <param name="assistant_id">The optional assistant identifier from the route.</param>
+    /// <param name="thread_id">The optional thread identifier from the route.</param>
+    /// <returns>An <see cref="IResult"/> representing the result of the operation.</returns>
+    [ApiKeyAuthorize]
+    private static async Task<IResult> CreateThreadAsync(
+        [FromServices] ICatalogService catalogService,
+        [FromServices] IProxyService proxyService,
+        [FromServices] IAssistantService assistantService,
+        HttpContext context,
+        [FromBody] JsonDocument? requestJsonDoc = null,
+        string? assistant_id = null,
+        string? thread_id = null
+    )
+    {
+        var requestPath = context.Request.Path.Value!.Split("/api/v1/").Last();
+        var requestContext = (RequestContext)context.Items["RequestContext"]!;
+
+        var deployments = await catalogService.GetEventAssistantEndpoint(requestContext.EventId);
+        var deployment = deployments.FirstOrDefault();
```

Comment:
I'd expect that the `GetEventAssistantEndpoint` returns the nullable value so you don't have to do additional unpacking here.